services/navigation/builtin/builtin.go

```diff
@@ -412,17 +443,23 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 					svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
 					continue
 				}
+				svc.logger.Warnf("failed to navigate to waypoint: %+v due to error %s", wp, err.Error())
+				continue
+			}
 
-				svc.logger.Infof("skipping waypoint %+v due to error while navigating towards it: %s", wp, err)
-				if err := svc.waypointReached(ctx); err != nil {
-					if svc.waypointIsDeleted() {
-						svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
-						continue
-					}
-					svc.logger.Infof("can't mark waypoint %+v as reached, exiting navigation due to error: %s", wp, err)
-					return
+			// mark the waypoint as reached if the motion service hit no error
+			if err = svc.waypointReached(ctx); err != nil {
+				// If waypointReached returned an error it indicates the store didn't
+				// record the waypoint as being reached, so we Close the nav service
+				// as otherwise we will try to continue navigating to a waypoint
+				// we have already reached.
+				svc.logger.Errorf("failed to update the waypoint db %+v, calling Close() on nav", wp)
+				if err = svc.Close(ctx); err != nil {
+					svc.logger.Errorf("navigation Close() returned an error", err)
```

Comment:
I think this deserves a colon at the end of the string

Reply:
I moved it to use errors.Wrap, which I think should add colons